### PR TITLE
IOS-606: Profile of a user - adapt the current controller to display switching bar  

### PR DESCRIPTION
**Pull request by pbruz**
### Ticket
[IOS-606](https://netguru.atlassian.net/browse/IOS-606)

### Task Description
Profile of a user - adapt the current controller to display switching bar

### CODE REVIEW

File: Inbbbox/Source Files/ViewControllers/ProfileShotsOrMembersViewController.swift [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+        collectionView.registerClass(LargeUserCollectionViewCell.self, type: .cell)
+        collectionView.registerClass(ProfileHeaderView.self, type: .header)
+
+        do { // hides bottom border of navigationBar
+            let currentColorMode = ColorModeProvider.current()
+            navigationController?.navigationBar.shadowImage = UIImage(color: currentColorMode.navigationBarTint)
+            navigationController?.navigationBar.setBackgroundImage(
+				UIImage(color: currentColorMode.navigationBarTint),
+                for: .default
+			)
```
Comment (Malibris) [inline Inbbbox/Source Files/ViewControllers/ProfileShotsOrMembersViewController.swift]: Lines indentation looks really strange. Is this GitHub formatting or something bad is going on?

File: Inbbbox/Source Files/ViewControllers/ProfileShotsOrMembersViewController.swift [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+            }
+            return cell
+        }
+
+        if let viewModel = viewModel as? TeamDetailsViewModel {
+            let cellData = viewModel.userCollectionViewCellViewData(indexPath)
+
+            indexPathsNeededImageUpdate.append(indexPath)
+
+            if collectionView.collectionViewLayout.isKind(of: TwoColumnsCollectionViewFlowLayout.self) {
```
Comment (Malibris) [inline Inbbbox/Source Files/ViewControllers/ProfileShotsOrMembersViewController.swift]: Consider moving a cell creation code for `TeamDetailsViewModel` into another method. Currently `cellForItemAt` is ~80 LOC.

File: Inbbbox/Source Files/ViewControllers/ProfileShotsOrMembersViewController.swift [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+                    cell.fourthShotImageView.loadImageFromURL(cellData.shotsImagesURLs![3])
+                }
+                if !cell.isRegisteredTo3DTouch {
+                    cell.isRegisteredTo3DTouch = registerTo3DTouch(cell.contentView)
+                }
+                return cell
+            } else {
+                let cell = collectionView.dequeueReusableClass(LargeUserCollectionViewCell.self,
+                                                               forIndexPath: indexPath, type: .cell)
+                cell.clearImages()
```
Comment (Malibris) [inline Inbbbox/Source Files/ViewControllers/ProfileShotsOrMembersViewController.swift]: Maybe move this code into the `prepareForReuse` method of the cell itself?

File: Inbbbox/Source Files/ViewModels/ProfileViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,104 @@
+//
+//  ProfileViewModel.swift
+//  Inbbbox
+//
+//  Created by Peter Bruz on 24/01/2017.
```
Comment (Malibris) [inline Inbbbox/Source Files/ViewModels/ProfileViewModel.swift]: Unnecessary line.

File: Inbbbox/Source Files/Views/Controllers View/ProfileView.swift [hunk 5]
```diff
@@ -0,0 +1,74 @@
+//
+//  ProfileView.swift
+//  Inbbbox
+//
+//  Created by Peter Bruz on 23/01/2017.
```
Comment (Malibris) [inline Inbbbox/Source Files/Views/Controllers View/ProfileView.swift]: Unnecessary line

File: Inbbbox/Source Files/ViewControllers/ProfileShotsOrMembersViewController.swift (hunk 1 above)
Comment (pbruz) [inline Inbbbox/Source Files/ViewControllers/ProfileShotsOrMembersViewController.swift]: GH messed it a little bit up. But I'll make it one-liner.

Comment (pbruz) [inline Inbbbox/Source Files/ViewControllers/ProfileShotsOrMembersViewController.swift]: Ouch, in fact it shouldn't be in this class 🙈  But I will fix it in other class.

File: Inbbbox/Source Files/ViewControllers/ProfileShotsOrMembersViewController.swift (hunk 2 above)
Comment (pbruz) [inline Inbbbox/Source Files/ViewControllers/ProfileShotsOrMembersViewController.swift]: Heh. It's the code that was written many months ago, I just had to move it to other class. But ok, I'll check what can be done here.

File: Inbbbox/Source Files/ViewControllers/ProfileShotsOrMembersViewController.swift (hunk 3 above)
Comment (pbruz) [inline Inbbbox/Source Files/ViewControllers/ProfileShotsOrMembersViewController.swift]: I didn't change it as well in the PR, but sounds reasonable - I'll apply 👍

**Pull request merged**